### ISSUE TITLE
Export alias neither is suppported nor throws error when building

### ISSUE THREAD

**Issue by udondan** [aws/jsii#2102]
### General Information
* **JSII Version:** any
* **Platform:**     any

### What is the problem?

```ts
export * as foo from './foo';
```

In subdir foo are some exported classes.

This compiles w/o errors, though the generated packages do not contain any contents from foo.

My assumption here is that an export alias (or whatever that `as foo` part is called) simply is not supported.

**Comment by RomainMuller** [comment]
This ought to be fully supported and is considered a submodule in the jsii type model. We even test this:

https://github.com/aws/jsii/blob/5701e4e4116e10ffacafa4930c0db23972f7e0e4/packages/jsii-calc/lib/index.ts#L10

At this stage I'ml going to be venturing there is something else causing problems that the compiler fails to check for... May I ask to see the code you have which causes this, or a minimal reproduction?

**Comment by udondan** [comment]
Sorry for the very late reply. I didn't work on this piece until now. 

The behavior seems to have changed in the meantime. It no longer passes without errors. Instead it throws this error:

> Exported APIs cannot use un-exported type "something.Something"

So I think this is kind of the same problem as #1818

In the submodule I have 

```typescript
export { Something } from './something.ts'
```

When I change this to

```typescript
export * from './something.ts'
```

it works. Though the only exported item in `something.ts` is `Something`...

I tried to produce a minimal example repo, but I failed. So I think you're also correct in the assumption that this is somehow caused by or in combination with something else.

I'll point you to the most complex example possible instead. 😈 This is about line 2 in [this file](https://github.com/udondan/iam-floyd/blob/a1ce5dcd8419345494927ad17c71a3305dd6b23b/lib/index.ts#L2). I want to change it to:

```typescript
export * as statement from './generated';
```

In the file [./generated/index.ts](https://github.com/udondan/iam-floyd/blob/a1ce5dcd8419345494927ad17c71a3305dd6b23b/lib/generated/index.ts) you see I have all the exports defined as

```typescript
export { A4b } from "./alexaforbusiness";
export { ExecuteApi } from "./amazonapigateway";
export { Appflow } from "./amazonappflow";
...
```

jsii stops complainig when I change them all to:

```typescript
export * from "./alexaforbusiness";
export * from "./amazonapigateway";
export * from "./amazonappflow";
...
```

Pinging @RomainMuller / @SomayaB Think you won't see this, because it's closed.
